### PR TITLE
Fix #13670: fix alias option when it includes `-`  or `_` in option name

### LINKED ISSUE #13670: Console optionAliases() using hyphen (-) mark on alias
I running console using options that I set on `optionAliases()` and assume I can using hyphen (-) mark. Example: `['start-date' => 'startDate']`.
So I set with `-start-date="2017-01-01"`

I check `yii\console\Request` only `^-(\w+)` allowed.
Since its name changed from `shortCuts` to `optionAliases`, I think using hyphen (-) should included. 

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 7.0.1

### PR DESCRIPTION
**Pull request by pana1990**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13670

any chars that should be include? i have included `_` and `-`

For example, the following statements are accept now : 

-date_start="2017-12-12",
-date-start="2017-12-12"

### CODE REVIEW

**Review by SilverFire — APPROVED** [review]

File: framework/console/Request.php [hunk 1]
```diff
@@ -76,7 +76,7 @@ public function resolve()
                 $params[] = $param;
             } elseif ($param === '--') {
                 $endOfOptionsFound = true;
-            } elseif (preg_match('/^--(\w+)(?:=(.*))?$/', $param, $matches)) {
+            } elseif (preg_match('/^--([a-zA-Z0-9-_]+)(?:=(.*))?$/', $param, $matches)) {
```
Comment (Kolyunya) [inline framework/console/Request.php]: Isn't `[a-zA-Z0-9-_]+` equal to the `[\w-]+`?

Comment (pana1990) [inline framework/console/Request.php]: yes, i am going to change it, thanks :)

File: framework/console/Request.php [hunk 2]
```diff
@@ -85,7 +85,7 @@ public function resolve()
                 if ($name !== Application::OPTION_APPCONFIG) {
                     $params[$name] = isset($matches[2]) ? $matches[2] : true;
                 }
-            } elseif (preg_match('/^-(\w+)(?:=(.*))?$/', $param, $matches)) {
+            } elseif (preg_match('/^-([a-zA-Z0-9-_]+)(?:=(.*))?$/', $param, $matches)) {
```
Comment (samdark) [inline framework/console/Request.php]: Same as https://github.com/yiisoft/yii2/pull/13678/files#r103426857

**Review by samdark — APPROVED** [review]

File: framework/console/Request.php (hunk 2 above)
Comment (pana1990) [inline framework/console/Request.php]: yep

Comment (pana1990) [inline framework/console/Request.php]: done!

File: framework/console/Request.php (hunk 1 above)
Comment (pana1990) [inline framework/console/Request.php]: done!

File: framework/console/Request.php [hunk 3]
```diff
@@ -76,7 +76,7 @@ public function resolve()
                 $params[] = $param;
             } elseif ($param === '--') {
                 $endOfOptionsFound = true;
-            } elseif (preg_match('/^--(\w+)(?:=(.*))?$/', $param, $matches)) {
+            } elseif (preg_match('/^--([\w-_]+)(?:=(.*))?$/', $param, $matches)) {
```
Comment (Kolyunya) [inline framework/console/Request.php]: Isn't `[\w-_]` equal to the `[\w-]`?

Comment (pana1990) [inline framework/console/Request.php]: yeah, done! 

thanks a lot :)

**Review by dynasource — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**